ui.apps/src/main/content/jcr_root/apps/core/fd/components/commons/v1/clientlibs/layouts/iteminjector/js/itemapi.js

```diff
@@ -0,0 +1,272 @@
+
+                if (data.tooltip) {
+                    linkElem.setAttribute("title", data.tooltip);
+                }
+            }
+            else {
+                item.classList.add(ERROR_CLASS);
+            }
+
+            if (data.thumbnailLink) {
+                thumbnailElem.setAttribute("src", data.thumbnailLink);
+            } else {
+                thumbnailElem.style.display = "none";
+            }
+
+            if (data.operations) {
+                operationsElem.setAttribute("operation_model_id", data.id);
+                operationsElem.setAttribute("operations", JSON.stringify(data.operations));
+                operationsElem.addEventListener('click',function (event) {
+                    menu.showHidePopover(operationsElem);
+                });
+            } else if (operationsElem) {
+                operationsElem.style.display = "none";
+            }
+
+            if (data.lastModified && timeInfoElem) {
+                addDateWithTooltip(timeInfoElem, data.lastModified);
+            }
+            itemContainer.appendChild(item);
+        },
+        init: function (config) {
+            var itemTemplate = config.itemTemplate,
+                container = config.resultsNode;
+
+            //add menuTemplate if there, initialize menu for component config
+            if (config.menuTemplate) {
+                config.menu = new Menu(config.menuTemplate, config.queryPath);
+                var el = document.createElement("span");
+                el.innerHTML = config.menuTemplate;
+                var menuInclude = el.querySelector(Menu.selectors.includes);
+                container.appendChild(menuInclude);
+            }
+
+            // Initialize container
+            var el = document.createElement("span");
+            el.innerHTML = itemTemplate;
+
+            var resultContainer = el.querySelector(selectors.container),
+                includeElements = el.querySelector(selectors.includes);
+
+            container.appendChild(includeElements);
+            container.appendChild(resultContainer);
+        },
+        clear: function (template, container) {
+            var el = document.createElement("span");
+            el.innerHTML = template;
+
+            var originalState = el.querySelector(selectors.container),
+                resultContainer = container.querySelector(selectors.container);
+
+            resultContainer.innerHTML = originalState.innerHTML;
+        }
+    };
+
+    var onScriptLoad = function (event, additionalParams) {
+        var componentApp = additionalParams.detail.portalLister;
+        componentApp.initializeItemAPI(ItemAPI);
+    };
+
+    $(window).on("core-forms-itemapi-onload", onScriptLoad);
+
+    var registerOperation = function (event, operationConfig) {
+        operationHandlers[operationConfig.name] = operationConfig.handler;
+    }
+
+    $(window).on("core-forms-register-operation", registerOperation);
```

Comment:
You would have to document this event in the component documentation

Reply:
Added